### DISCUSSION TITLE
Piano LED Visualizer et Synthesia

### DISCUSSION THREAD

**Discussion by MymyOuah** [onlaj/Piano-LED-Visualizer#524] [Q&A]
Good evening, I’ve just installed Piano Led Visualizer on my synthesizer, and I would like the notes to light up progressively with the Synthesia app. At the moment, the connection between Synthesia and Piano Led Visualizer works fine, as I can start a song, and the note lights up correctly when I press it, but I don’t have the anticipation of the upcoming notes. Do you know where the issue might come from? What settings should I adjust? For your information, I’ve tried both from the Raspberry Pi and from the Synthesia app, but I can’t pinpoint the problem. Thank you in advance for your help. Have a great evening.

**Comment by TheGuyFawkes** [comment]
> Good evening, I’ve just installed Piano Led Visualizer on my synthesizer, and I would like the notes to light up progressively with the Synthesia app. At the moment, the connection between Synthesia and Piano Led Visualizer works fine, as I can start a song, and the note lights up correctly when I press it, but I don’t have the anticipation of the upcoming notes. Do you know where the issue might come from? What settings should I adjust? For your information, I’ve tried both from the Raspberry Pi and from the Synthesia app, but I can’t pinpoint the problem. Thank you in advance for your help. Have a great evening.

After a lot of struggle I managed to get it working, I don't know if I did it "my way". What I do is upload the same MIDI file to Synthesia and to the "Songs" section of Visualizer. 
![17318326664011437118618647253715](https://github.com/user-attachments/assets/c1f552a3-779b-42ac-aab5-5cc57a98a649)

Once I have both files uploaded to both platforms, I go into Visualizer and click on the student hat icon for the song (in this case for Elise). 

![17318328716744850850193373695569](https://github.com/user-attachments/assets/5071dd57-ddfa-4a8d-9155-996245151fc3)
Y después a start learning:
![17318329583166456558682783404834](https://github.com/user-attachments/assets/60b600dc-bc96-4d5f-8999-904a17854080)
Then the lights come on:
![17318329911778214050477829206285](https://github.com/user-attachments/assets/00134d2c-0668-4971-bf6c-ab0be1452c28)

Now I go into Synthesia and put the same MIDI file:

https://github.com/user-attachments/assets/f9ca2b35-a4e7-4b85-bad8-e74308fff9e3

As a "workaround" I hope it helps. If you find a way to do it natively using only Synthesia please let me know!

**Comment by onlaj** [comment]
> but I don’t have the anticipation of the upcoming notes

@MymyOuah 
It doesn't light up the next-to-play key? Or it doesn't wait for your input?
If it's the former, you need to change the input to your PC (either rtp-midi connection or Sevilla's USB device).
If it's the latter, you need to change mode to practice in the Synthesia.
